.github/workflows/breaking-change-alert.yaml

```diff
@@ -0,0 +1,122 @@
+name: Breaking Change Notifications
+
+on:
+  workflow_call:
+    inputs:
+      sender_login:
+        description: 'Sender login'
+        required: true
+        type: string
+      sender_avatar:
+        description: 'Sender avatar URL'
+        required: true
+        type: string
+      repo:
+        description: 'Repository'
+        required: true
+        type: string
+      pr_number:
+        description: 'Pull request number'
+        required: true
+        type: number
+      pr_title:
+        description: 'Pull request title'
+        required: true
+        type: string
+      pr_body:
+        description: 'Pull request body'
+        required: false
+        default: '_Empty PR description_'
+        type: string
+      pr_base_ref:
+        description: 'Base branch of the pull request'
+        required: true
+        type: string
+      pr_author:
+        description: 'Pull request author'
+        required: true
+        type: string
+      event_action:
+        description: 'Event action'
+        required: true
+        type: string
+      pr_merged:
+        description: 'Pull request merged status'
+        required: true
+        type: boolean
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  notify-breaking-changes:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Determine notification parameters
+        id: notification
+        run: |
+          if [[ "${{ inputs.event_action }}" == "closed" && "${{ inputs.pr_merged }}" == "true" ]]; then
+            echo "action=Merged" >> $GITHUB_OUTPUT
+            echo "color=#D00000" >> $GITHUB_OUTPUT
+            echo "icon=:rocket:" >> $GITHUB_OUTPUT
+          elif [[ "${{ inputs.event_action }}" == "closed" ]]; then
+            echo "action=Closed" >> $GITHUB_OUTPUT
+            echo "color=#1d9bd1" >> $GITHUB_OUTPUT
+            echo "icon=:information_source:" >> $GITHUB_OUTPUT
+          elif [[ "${{ inputs.event_action }}" == "reopened" ]]; then
+            echo "action=Reopened" >> $GITHUB_OUTPUT
+            echo "color=warning" >> $GITHUB_OUTPUT
+            echo "icon=:warning:" >> $GITHUB_OUTPUT
+          else
+            action=${{ inputs.event_action }}
+            echo "action=${action^}" >> $GITHUB_OUTPUT
+            echo "color=good" >> $GITHUB_OUTPUT
+            echo "icon=:information_source:" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Send Slack notification
+        uses: slackapi/slack-github-action@v1.27.0
+        with:
+          payload: |
+            {
+              "icon_emoji": "${{ steps.notification.outputs.icon }}",
+              "username": "RAPIDS Breaking Change Notifier - ${{ steps.notification.outputs.action }}",
+              "attachments": [
+                {
+                  "author_name": "${{ steps.notification.outputs.action }} PR by ${{ inputs.sender_login }}",
```

Comment:
We should use the step above (or add a new step) to sanitize `inputs.*`, or at the very least escape them for JSON.

Reply:
@KyleFromNVIDIA I presume that AJ's suggestion to move all inputs properties to intermediate environment variables should perhaps suffice for catering to injection attacks.

Do you think we should still sanitize after doing that?

Reply:
Environment variable substitution won't work in this step. This isn't a script, it's a JSON input that gets parsed by the external action. Someone could manipulate `inputs.sender_login` to make the external action do Bad Things. The way to mitigate that is to pass `inputs.sender_login` to a prior step, escape it as a JSON string, and then pass the escaped version to this, perhaps as `steps.sanitize.outputs.sender_login`.

Reply:
I see what you're referring to now, thanks!